specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimapis.json

```diff
@@ -0,0 +1,1878 @@
+            "required": false,
+            "type": "string",
+            "description": "| Field       | Supported operators    | Supported functions               |\n|-------------|------------------------|-----------------------------------|\n| id          | ge, le, eq, ne, gt, lt | substringof, startswith, endswith |\n| name        | ge, le, eq, ne, gt, lt | substringof, startswith, endswith |\n| description | ge, le, eq, ne, gt, lt | substringof, startswith, endswith |\n| serviceUrl  | ge, le, eq, ne, gt, lt | substringof, startswith, endswith |\n| path        | ge, le, eq, ne, gt, lt | substringof, startswith, endswith |"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Paged Result reponse of Apis.",
+            "schema": {
+              "$ref": "#/definitions/ApiCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/ApiContract"
+      }
+    },
+    "/apis/{apiId}": {
+      "get": {
+        "tags": [
+          "Apis"
+        ],
+        "operationId": "Api_Get",
+        "description": "Gets the details of the API specified by its identifier.",
+        "x-ms-examples": {
+          "ApiManagementGetApis": {
+            "$ref": "./examples/ApiManagementGetApis.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified API entity.",
+            "schema": {
+              "$ref": "#/definitions/ApiContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Apis"
+        ],
+        "operationId": "Api_CreateOrUpdate",
```

Comment:
>Api_CreateOrUpdate [](start = 24, length = 18)

Is this long running operation?

Reply:
no

---
In reply to: [163341446](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163341446) [](ancestors = 163341446)